web/src/app/escalation-policies/PolicyStepFormDest.stories.tsx

```diff
@@ -0,0 +1,130 @@
+              destinationFieldValidate:
+                vars.input.value === 'https://target.com' ||
+                vars.input.value === 'https://target.com2' ||
+                vars.input.value === '+12225558989',
+            },
+          })
+        }),
+
+        graphql.query('DestDisplayInfo', ({ variables: vars }) => {
+          let body = {}
+          if (vars.input.values[0].value === 'https://target2.com') {
+            body = {
+              data: {
+                destinationDisplayInfo: {
+                  text: vars.input.values[0].value,
+                  iconURL: 'builtin://webhook',
+                  iconAltText: 'Webhook',
+                },
+              },
+            }
+          } else {
+            body = {
+              data: {
+                destinationDisplayInfo: {
+                  text: vars.input.values[0].value,
+                  iconURL: 'builtin://phone-voice',
+                  iconAltText: 'Voice Call',
+                },
+              },
+            }
+          }
+          return HttpResponse.json(body)
+        }),
+      ],
+    },
+  },
+} satisfies Meta<typeof PolicyStepFormDest>
+
+export default meta
+type Story = StoryObj<typeof meta>
+
+export const AddActions: Story = {
+  args: {
+    value: {
+      delayMinutes: 15,
+      actions: [],
+    },
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement)
+
+    // add valid phone number and webhook
+    await waitFor(async () => {
+      await userEvent.type(
+        await canvas.findByPlaceholderText('11235550123'),
+        '12225558989',
+      )
+    })
+    await expect(await canvas.findByTestId('CheckIcon')).toBeVisible()
+
+    await userEvent.clear(
+      await canvas.findByPlaceholderText('https://example.com'),
+    )
+    await waitFor(async () => {
+      await userEvent.type(
+        await canvas.findByPlaceholderText('https://example.com'),
+        'https://target.com',
+      )
+    })
+
+    await userEvent.click(await canvas.findByText('Add Action'))
+
+    // expect to see action added
+    await expect(await canvas.findByTestId('destination-chip')).toBeVisible()
+    await expect(await canvas.findByText('+12225558989')).toBeVisible()
+
+    // add single destination action
+    await userEvent.click(await canvas.findByText('Multi Field EP Step Dest'))
+    await userEvent.click(await screen.findByText('Single Field EP Step Dest'))
+    await userEvent.clear(
+      await canvas.findByPlaceholderText('https://example.com'),
+    )
+    await waitFor(async () => {
+      await userEvent.type(
+        await canvas.findByPlaceholderText('https://example.com'),
+        'https://target2.com',
+      )
+    })
+
+    await userEvent.click(await canvas.findByText('Add Action'))
+
+    // expect to see action added
+    await expect(await canvas.findByText('https://target2.com')).toBeVisible()
+    await expect(await canvas.findByTestId('WebhookIcon')).toBeVisible()
+  },
```

Comment:
I feel like this is really just testing the DestField is working.

I'd like to see an interaction test for:
- Click Add Action, and show an error on dest type, the field, and a generic error (i.e., unknown, to the left of the button)
- Editing an input field clears the errors
- With no validation error, Add Action adds the chip to the list and clears it's form

Reply:
I was able to add assertions for all of these, except for showing an error on the destination type and a generic error (one of these is covered in https://github.com/target/goalert/pull/3718). This PR already consists of 440 lines of code changes. We should consider increasing the PR size threshold if we want to add these extra assertions because attempting to add all these test cases and trying to stay under 500 lines of code change has been a (seemingly unnecessary) challenge.